modules/standard/Socket.md

```diff
@@ -0,0 +1,82 @@
+## Socket Module API
+
+To those familiar with the Unix socket API, the method names will feel familiar, though their usage will be somewhat simpler than the raw Unix socket API.
+
+## Creating a Listener
+
+`socket.listen` function will create a `TCPListener` Instance which will take in following parameters
+
+```chapel
+socket.listen([addr:IPAddr,]port:int,backlog=5,reuseaddr=true)
+```
+
+The `TCPListener` instance will have methods on it to call
+- `accept`,
+- `close`,
+- `addr` and
+- `setSocketOpt` to set various socket options.
+
+`accept` method will return a `TCPConn` instance and associated `addr` of the connection.
+
+## Connecting to remote sockets
+
+`socket.connect` function will create `TCPConn` instance which will take in the following parameters
```

Comment:
>I have defined `TCPConn` as a translation of `UDPSocket` because the socket isn't just in an unconnected state but will have an established connection and also helps to differ from the TCP socket one we get from `listen`

I think we can discuss more what it should be named while `TCPSocket` is the one Julia uses in my opinion it correlates to just a normal socket whereas here this socket has established connection which is what Go uses